lib/_stream_readable.js

```diff
@@ -561,7 +561,7 @@ function maybeReadMore_(stream, state) {
 // for virtual (non-string, non-buffer) streams, "length" is somewhat
 // arbitrary, and perhaps not very meaningful.
 Readable.prototype._read = function(n) {
-  this.emit('error', new errors.Error('ERR_STREAM_READ_NOT_IMPLEMENTED'));
+  throw new errors.Error('ERR_METHOD_NOT_IMPLEMENTED', '_read()');
```

Comment:
I would leave this as `this.emit('error')`.

Reply:
Can you elaborate on the reasons?